addons/stock_picking_batch/models/stock_move_line.py

```diff
@@ -67,7 +67,10 @@ def _add_to_wave(self, wave=False):
                     picking_to_wave_vals['move_lines'] += [Command.link(move.id)]
                     continue
                 # Split the move
-                qty = sum(move_lines.mapped('product_qty'))
+                if move.from_immediate_transfer:
+                    qty = sum(move_lines.mapped('qty_done'))
```

Comment:
`move.quantity_done` to have the uom conversion

Reply:
modified in following commit anyway